src/Model/IdentifiableEntity.java

```diff
@@ -0,0 +1,15 @@
+package Model;
+
+public abstract class IdentifiableEntity {
+    protected static int counter = 0;
```

Comment:
why do you choose this access modifier?

Reply:
> why do you choose this access modifier?
To have the ability to use this field in child classes